src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const lastElement = this[this.length - 1];
+    let string = '';
+
+    for (const element of this) {
+      const elem = element !== null && element !== undefined
+        ? element
+        : '';
+
+      string += element !== lastElement
```

Comment:
This checks for the equality with lastElement. In case we have 
```js
['asd', '1', '2', 'asd', 'asd'].join2(',');
```

I believe the code would result 
```js
asd1,2,asdasd
```

Consider looping through indexes:
```js
for (let i = 0 ...

Reply:
You're right, index would be a more reliable comparison